031-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,168 @@
+# TLS encrypting the Kafka Connect REST API
+
+This proposal adds the option to reduce access to the Kafka Connect REST API by enabling TLS encryption on the Kafka Connect REST listener.
+
+## Current situation
+
+Currently, instances of Kafka Connect that are deployed by the Strimzi operators are configured with the default REST API endpoint settings.
+This means that the Kafka Connect REST API endpoint uses HTTP on port 8083 and that the Strimzi `KafkaConnect`, `KafkaConnector` and `KafkaMirrorMaker2` operators make unsecured REST client calls based on this default configuration.
+
+The default network policies created by the Strimzi operators restrict incoming REST API calls to only allow access from the operator pod.
+Users can define further network policies to override the default policy and allow wider access.
+
+## Motivation
+
+There is currently no way to TLS encrypt the Kafka Connect REST API, leaving the Connect REST listener as perhaps the last key endpoint that cannot currently be restricted using Strimzi.
+
+## Proposal
+
+This proposal adds REST api configuration options to the `KafkaMirrorMaker2` and `KafkaConnect` CRDs.
+
+A new field is added to the `KafkaConnect` and `KafkaMirrorMaker2` specs named `restListeners`, which allows users to configure an HTTP listener on port 8083, an encrypted HTTPS listener on port 8443, or both in the Kafka Connect configuration.
+This enables users to access the REST API with TLS if required.
+The `KafkaConnector`, `KafkaConnect` and `KafkaMirrorMaker2` operators will use the TLS encrypted listener when it is enabled, even if an unencrypted listener is present.
+
+The `restListeners` field is designed for potential future extension, and is a list of items that contains a `protocol` field and an optional `restCertChainAndKey` field.
+
+For this proposal, the `protocol` field only supports two values: `http` and `https`. If the `restListeners` list is empty the `KafkaConnect` and `KafkaMirrorMaker2` 
```

Comment:
I wonder if this should be `type` instead of `protocol` ... but really just 🤔  about it ... not sure what is better.

Reply:
I think we need restListeners to be an object rather than a list because we need a field to specify the certificates

Reply:
Would `type` be confusing given that the listeners has a `type` field but that is `internal` vs `route` vs `loadbalancer`. Here we aren't specifying the way the listener is exposed in a Kubernetes sense, but just whether it is secure. We could instead of having `protocol` have a boolean field called `tls`. That would match the listeners, but wouldn't be extensible. Although I can't think of what other protocols the Connect rest listeners might support